### ISSUE TITLE
Make `butter` and `filtfilt` parameters accessible to users in `clean`

### ISSUE THREAD

**Issue by tsalo** [nilearn/nilearn#3475]
For `scipy.signal.butter`, the `order` parameter is accessible in the nilearn function `signal.butterworth`, but it isn't accessible in `signal.clean`.

For `scipy.signal.filtfilt`, the `padtype`, `padlen`, `method`, and `irlen` parameters aren't accessible through nilearn. In particular, `padtype` and `padlen` would be very nice to set in `signal.clean`.

### Benefits to the change
I don't know if the default `padtype` ("odd"), or the hardcoded `order` (5) are well-suited to BOLD data, and I haven't come across any direct comparisons of padding types in the literature, so it seems like something users should have control over.

I've also done some informal checks with some data I had on hand, and the `odd` option that nilearn uses seems to show some edge effects that the `constant` option, for example, doesn't.

### Pseudocode for the new behavior, if applicable

```python
import numpy as np

from nilearn import signal

data = np.random.random((100, 10000))
denoised_data = signal.clean(
    data,
    filter="butterworth",
    low_pass=0.1,
    high_pass=0.01,
    padtype="constant",  # or "odd", or "even"
    filter_order=6,  # fed into scipy.signal.butter as "N"
    t_r=2,
)
```

**Comment by bthirion** [comment]
There are two aspects in this issue:
* some parameters are hidden. This is not great, but adding 4 parameters to clean make the API substantially more complex. Add them as  **kwargs ?
* current parameter setting may be suboptimal. Indeed, we should improve the default setting.

**Comment by ymzayek** [comment]
I also agree to include them as **kwargs if we decide to include them all but maybe it's not necessary? I am not sure about the relative importance of each parameter mentioned but it would also be something to look into the impact of each.

+1 for optimizing the current settings

@htwangtw do you have any input on this?

**Comment by tsalo** [comment]
Setting these parameters with `kwargs` (possibly with prefixes to extract relevant parameters for relevant functions) sounds like a good idea to me.

> I am not sure about the relative importance of each parameter mentioned but it would also be something to look into the impact of each.

Unfortunately, it does look like different padding types can have an impact at the beginning and end of time series. I tried comparing the different options, including filtering without padding and zero-padding (since that's what comes up most often in the literature, AFAICT):

![filter_results](https://user-images.githubusercontent.com/8228902/215501975-4893a124-514e-4feb-9075-5c7da1ede394.png)

Here's the code I used (extension changed from .py to .txt so I could upload it):  [test_filters.txt](https://github.com/nilearn/nilearn/files/10536571/test_filters.txt). It's not exactly a comprehensive comparison, but I don't even know what to look for to determine the winner, beyond penalizing major peaks or troughs at the beginning and end of the time series. In that sense, the default (odd padding) doesn't seem to do well on this voxel's data.

I couldn't find any rationale online for scipy's padtype options, nor could I find any fMRI papers that mention any of those options, so it's not clear to me which is best suited to BOLD data. Additionally, I _have_ seen a few fMRI papers that mention zero-padding, but I can't tell if they used zero-padding for a good reason, or just because it's easy and that's what they'd heard of.
